test/integration/authz_v1_test.go

```diff
@@ -0,0 +1,388 @@
+package integration
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/yahoo/athenz/clients/go/zms"
+	"github.com/yahoo/k8s-athenz-istio-auth/pkg/istio/rbac/common"
+	"github.com/yahoo/k8s-athenz-istio-auth/test/integration/fixtures"
+	"github.com/yahoo/k8s-athenz-istio-auth/test/integration/framework"
+
+	"istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type action int
+
+const (
+	create action = iota
+	update
+	noop
+)
+
+// rolloutAndValidate will create / update / noop the athenz domain resource and wait for the
+// associated service role / service role bindings to be created. Once these are rolled out,
+// they are validated against the expected output.
+func rolloutAndValidate(t *testing.T, r *fixtures.ExpectedResources, a action) {
+	if a == update {
+		currentAD, err := framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Get(r.AD.Name, v1.GetOptions{})
+		assert.Nil(t, err, "error should be nil")
+		r.AD.ResourceVersion = currentAD.ResourceVersion
+		_, err = framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Update(r.AD)
+		assert.Nil(t, err, "error should be nil")
+	} else if a == create {
+		_, err := framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Create(r.AD)
+		assert.Nil(t, err, "should be nil")
+	}
+
+	err := wait.PollImmediate(time.Second, time.Second*5, func() (bool, error) {
+		for _, curr := range r.ModelConfigs {
+			got := framework.Global.IstioClientset.Get(curr.Type, curr.Name, curr.Namespace)
+			if got == nil {
+				return false, nil
+			}
+
+			if !reflect.DeepEqual(curr.Spec, got.Spec) {
+				return false, nil
+			}
+		}
+
+		return true, nil
+	})
+
+	if err != nil {
+		t.Error("time out waiting for rollout for ad", r.AD.Name, "with error", err)
+	}
+
+	validateConfigs(t, r)
+}
+
+// validateConfigs will validate the service role / service role bindings on the
+// cluster against the expected output
+func validateConfigs(t *testing.T, r *fixtures.ExpectedResources) {
+	for _, curr := range r.ModelConfigs {
```

Comment:
could we also validate if there are no other resources expect the ones expected in `r.ModelConfigs`?

Reply:
Yep, I added a check.